sbnana/SBNAna/NuMINumuXSec/ICARUSNumuXsec_HistoProducer.C

```diff
@@ -0,0 +1,561 @@
+  Spectrum *sNeutrinoQEResidualFraction = new Spectrum(loader, axNeutrinoQEResidualFraction, spillCut, cut);
+
+  cout << "[HistoProducer::bookRecoNeutrino] Saving spectrums" << endl;
+
+  map_cutName_to_vec_Spectrums[currentCutName].push_back(sNeutrinoCombinedEnergy);
+  map_cutName_to_vec_Spectrums[currentCutName].push_back(sNeutrinoTestEnergy);
+  map_cutName_to_vec_Spectrums[currentCutName].push_back(sNeutrinoQE);
+  map_cutName_to_vec_Spectrums[currentCutName].push_back(sNeutrinoCombinedEnergyResidual);
+  map_cutName_to_vec_Spectrums[currentCutName].push_back(sNeutrinoQEResidual);
+  map_cutName_to_vec_Spectrums[currentCutName].push_back(sNeutrinoCombinedEnergyResidualFraction);
+  map_cutName_to_vec_Spectrums[currentCutName].push_back(sNeutrinoQEResidualFraction);
+
+  if(ISysts.size()>0){
+
+    EnsembleSpectrum *esNeutrinoCombinedEnergy = new EnsembleSpectrum(loader, axNeutrinoCombinedEnergy, spillCut, cut, systWs);
+    EnsembleSpectrum *esNeutrinoQE = new EnsembleSpectrum(loader, axNeutrinoQE, spillCut, cut, systWs);
+
+		map_cutName_to_vec_EnsembleSpectrums[currentCutName].push_back(esNeutrinoCombinedEnergy);
+		map_cutName_to_vec_EnsembleSpectrums[currentCutName].push_back(esNeutrinoQE);
+  }
+
+  cout << "[HistoProducer::bookRecoNeutrino] Finished" << endl;
+
+}
+
+void HistoProducer::saveHistograms(){
+
+  outputfile->cd();
+
+  cout << "[HistoProducer::saveHistograms] Number of cuts = " << vec_cutNames.size() << endl;
+  const unsigned int nCutName = vec_cutNames.size();
+  if(nCutName>0){
+
+    TH1D *hist_CutNames = new TH1D("hist_CutNames", "", nCutName, 0., 1.*nCutName);
+
+    for(unsigned int ic=0; ic<nCutName; ic++){
+
+      TString cutName = vec_cutNames.at(ic);
+      TString dirName = "cutName"+TString::Itoa(ic,10);
+      TDirectory *dir = outputfile->GetDirectory(dirName);
+      if(!dir){
+        outputfile->mkdir(dirName);
+      }
+      outputfile->cd(dirName);
+
+      vector<Spectrum *> vec_Spectrums = map_cutName_to_vec_Spectrums[cutName];
+      vector<EnsembleSpectrum *> vec_EnsembleSpectrums = map_cutName_to_vec_EnsembleSpectrums[cutName];
+
+      //==== BinContetn = number of Spectrum
+      //==== BinError = number of EnsembleSpectrum
+      hist_CutNames->SetBinContent(ic+1, vec_Spectrums.size());
+      hist_CutNames->SetBinError(ic+1, vec_EnsembleSpectrums.size());
+      hist_CutNames->GetXaxis()->SetBinLabel(ic+1, cutName);
+
+      cout << "[HistoProducer::saveHistograms] cutName = " << cutName << endl;
+      cout << "[HistoProducer::saveHistograms]   Directory name = " << dirName << endl;
+
+      //==== Spectrum
+      cout << "[HistoProducer::saveHistograms]   Number of Spectrum = " << vec_Spectrums.size() << endl;
+      for(unsigned int i=0; i<vec_Spectrums.size(); i++){
+
+        cout << "[HistoProducer::saveHistograms]   " << i << "-th Spectrum.." << endl;
+
+        if(vec_Spectrums.at(i)->GetBinnings().size()==1){
+          TH1 *h = vec_Spectrums.at(i)->ToTH1(TargetPOT);
```

Comment:
You can save an actual `Spectrum` object with `spect.SaveTo(dir->mkdir("name"))`. The advantage over saving just the histogram is that it saves the POT too, and you can load it back as a `Spectrum` object later.